kratos/utilities/data_type_traits.h

```
@@ -806,6 +1031,37 @@ template<class TDataType> class DataTypeTraits<DenseVector<TDataType>>
         }
     }
 
+    /**
+     * @brief Copies the given container element value to contiguous array.
+     *
+     * This method copies all the elements of @ref rContainer recursively to the
+     * provided @ref pContiguousDataBegin.
+     *
+     * @warning This may seg-fault if the the contiguous array given by @ref pContiguousDataBegin
+     *          is not correctly sized.
+     *
+     * @param pContiguousDataBegin      Contiguous array pointer.
+     * @param rContainer                Container to copy data to the contiguous array.
+     * @param pShapeBegin               Begining of the shape array.
+     * @param pShapeEnd                 End of the shape array.
+     */
+    template<class TIteratorType>
+    inline static void CopyToContiguousData(
+        PrimitiveType* pContiguousDataBegin,
+        const ContainerType& rContainer,
+        TIteratorType pShapeBegin,
+        TIteratorType pShapeEnd)
+    {
+        // since the range [pShapeBegin, pShapeEnd) is checked using the IsValidShape method,
+        // here we only check for the dynamic size.
+        if (*pShapeBegin <= rContainer.size()) {
```

if this condition is not met, than you are simply skipping the loop.

somewhere we should throw an error ...

---

Good idea, I will do that :)

---

done :)